### PR TITLE
Add failing test to reproduce bug with associative arrays

### PR DESCRIPTION
**Pull request by lcobucci**
For some weird reason travis didn't run on https://github.com/doctrine/coding-standard/pull/102, reopening it then...

### CODE REVIEW

File: tests/expected_report.txt [hunk 1]
```diff
@@ -24,13 +24,14 @@ tests/input/return_type_on_methods.php                17      0
 tests/input/semicolon_spacing.php                     3       0
 tests/input/static-closures.php                       1       0
 tests/input/test-case.php                             8       0
+tests/input/trailling_comma_on_array.php              1       0
```
Comment (alcaeus) [inline tests/expected_report.txt]: Typo: trailing 🤓

Also, the file name should also use dashes for consistency with other file names.

Comment (vladyslavstartsev) [reply tests/expected_report.txt]: > Typo: trailing 
> 
> Also, the file name should also use dashes for consistency with other file names.

that's mine

Comment (vladyslavstartsev) [reply tests/expected_report.txt]: I was thinking if I need to use dashes or underlines

Comment (alcaeus) [reply tests/expected_report.txt]: Right. Might as well leave it as long as the typo is fixed :)

**Comment by lcobucci** [comment]
@Ocramius this should go to `5.0.1`, no?

File: lib/Doctrine/ruleset.xml [hunk 2]
```diff
@@ -375,6 +375,7 @@
         <exclude name="Squiz.Arrays.ArrayDeclaration.ValueNotAligned"/>
         <!-- Checked by SlevomatCodingStandard.Arrays.TrailingArrayComma.MissingTrailingComma -->
         <exclude name="Squiz.Arrays.ArrayDeclaration.NoCommaAfterLast"/>
+        <exclude name="Squiz.Arrays.ArrayDeclaration.NoComma"/>
```
Comment (carusogabriel) [inline lib/Doctrine/ruleset.xml]: Maybe a comment explaining the reason?

Comment (lcobucci) [reply lib/Doctrine/ruleset.xml]: The comment is the one above :smile: (L376)

File: tests/fixed/trailing_comma_on_array.php [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+declare(strict_types=1);
+
+$array =  [
```
Comment (carusogabriel) [inline tests/fixed/trailing_comma_on_array.php]: Now that I noticed, we don't have a rule to force only 1 space after the assignment operator 🤔

Comment (vladyslavstartsev) [reply tests/fixed/trailing_comma_on_array.php]: ha! good point, I didn't notice that I've put 2 of them here

Comment (lcobucci) [reply tests/fixed/trailing_comma_on_array.php]: I didn't notice that as well :scream:

**Pull request merged**